WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -473,10 +473,10 @@ extension ProductFormViewModel {
                     }
                     self.resetProduct(data.product)
                     self.resetPassword(data.password)
-                    onCompletion(.success(data.product))
                     self.replaceProductID(productIDBeforeSave: productIDBeforeSave)
                     self.saveProductImagesWhenNoneIsPendingUploadAnymore()
-                    self.onProductCreated()
+                    self.onProductCreated(data.product.product)
+                    onCompletion(.success(data.product))
```

Comment:
Moving `onCompletion` to the end ensures the two modals (the product form and the "Saving your product" modal) are dismissed at the same time, for a smoother animation.